core/modules/system/system.menu.inc

```diff
@@ -146,6 +147,17 @@ function system_menu_block_form($config) {
       '!menu_settings' => $link,
     )),
   );
+  $form['toggle_text'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Toggle button text'),
+    '#default_value' => $config['toggle_text'],
+    '#description' => t('Text to display beside the toggle button. Default: <em>Menu</em>.'),
+    '#states' => array(
```

Comment:
Lets also add the relatively new `#indentation` property here, to nest the textfield under the "Display menu toggle button on small screens" checkbox:
```suggestion
    '#indentation' => 1,
    '#states' => array(
```